packages/salesforcedx-vscode-core/src/conflict/metadataCacheService.ts

```diff
@@ -278,17 +281,67 @@ export class MetadataCacheService {
       const cacheComponent = cacheIndex.get(key);
       const projectComponent = projectIndex.get(key);
       if (cacheComponent && projectComponent) {
-        components.push({
-          cacheComponent,
-          projectComponent,
-          lastModifiedDate: fileProperties.lastModifiedDate
-        });
+        if (cacheComponent.component && projectComponent.component) {
+          components.push({
+            cacheComponent: cacheComponent.component,
+            projectComponent: projectComponent.component,
+            lastModifiedDate: fileProperties.lastModifiedDate
+          });
+        } else {
+          cacheComponent.children.forEach((cacheChild, childKey) => {
+            const projectChild = projectComponent.children.get(childKey);
+            if (projectChild) {
+              components.push({
+                cacheComponent: cacheChild,
+                projectComponent: projectChild,
+                lastModifiedDate: fileProperties.lastModifiedDate
+              });
+            }
+          });
+        }
       }
     });
 
     return components;
   }
 
+  /**
+   * Creates a map in which parent components and their children are stored together
+   * @param index The map which is mutated by this function
+   * @param components The parent and/or child components to add to the map
+   */
+  private static pairParentsAndChildren(index: Map<string, RecomposedComponent>, components: SourceComponent[]) {
+    for (const comp of components) {
+      const key = MetadataCacheService.makeKey(comp.type.name, comp.fullName);
+      // If the component has a parent it is assumed to be a child
+      if (comp.parent) {
+        const parentKey = MetadataCacheService.makeKey(comp.parent.type.name, comp.parent.fullName);
+        const parentEntry = index.get(parentKey);
+        if (parentEntry) {
+          // Add the child component if we have an entry for the parent
+          parentEntry.children.set(key, comp);
+        } else {
+          // Create a new entry that does not have a parent yet
+          index.set(parentKey, {
+            children: new Map<string, SourceComponent>().set(key, comp)
+          });
+        }
+      } else {
+        const entry = index.get(key);
```

Comment:
Would the component (since it's not a child component) not already be associated with the entry in the index?